plugins/modules/inventory_intent.py

```diff
@@ -961,10 +1089,162 @@ def get_diff_merged(self, config):
         device_type = self.config[0].get("type", "NETWORK_DEVICE")
         device_resynced = self.config[0].get("device_resync", False)
         device_updated = self.config[0].get("device_updated", False)
+        device_reboot = self.config[0].get("reboot_device", False)
         self.result['log'] = []
 
+        if device_reboot:
+            device_ips = self.config[0].get("ip_address")
+            ap_mac_address_list = []
+            # get and store the apEthernetMacAddress of given devices
+            for device_ip in device_ips:
+                response = self.dnac._exec(
+                    family="devices",
+                    function='get_device_list',
+                    params={"managementIpAddress": device_ip}
+                )
+                response = response.get('response')[0]
+                if response['apEthernetMacAddress'] is not None:
+                    ap_mac_address_list.append(response['apEthernetMacAddress'])
+
+            if len(ap_mac_address_list) == 0:
+                self.status = "failed"
+                self.result['changed'] = False
+                msg = "Cannot find the AP devices for Rebooting"
+                self.log(msg)
+                self.msg = msg
+                return self
+
+            # Now call the Reboot Access Point API
+            reboot_params = {
+                "apMacAddresses": ap_mac_address_list
+            }
+            response = self.dnac._exec(
+                family="wireless",
+                function='reboot_access_points',
+                op_modifies=True,
+                params=reboot_params,
+            )
+            self.log(str(response))
+
+            if response and isinstance(response, dict):
+                task_id = response.get('response').get('taskId')
+                while True:
+                    execution_details = self.get_task_details(task_id)
+                    if 'url' in execution_details.get("progress"):
+                        self.status = "success"
+                        self.result['changed'] = True
+                        self.result['response'] = execution_details
+                        break
+                    elif execution_details.get("isError") and execution_details.get("failureReason"):
```

Comment:
we can move execution_details.get("failureReason") inside this elif. You can follow the above comments for failure reason

Reply:
changed.